app/assets/javascripts/spree/frontend/affirm_checkout.js

```diff
@@ -0,0 +1,74 @@
+(function(){
+  /* only include this setup once */
+  if (!window.AffirmPaymentMethods) {
+
+    /*****************************************************\
+        set the shared checkout data
+    \*****************************************************/
```

Comment:
Sure, that's the main part of my review request. Want to cleanup the js and write the best architecture possible here.